tools/run_executable.py

```diff
@@ -0,0 +1,132 @@
+
+        if sys.platform == "win32":
+            paks_dir = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(args.executable)), "..", "..", "Content", "Paks"))
+        elif sys.platform == "darwin":
+            paks_dir = os.path.realpath(os.path.join(args.executable, "Contents", "UE4", "SpearSim", "Content", "Paks"))
+        elif sys.platform == "linux":
+            paks_dir = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(args.executable)), "SpearSim", "Content", "Paks"))
+        else:
+            assert False
+
+        assert os.path.exists(paks_dir)
+
+        # we don't use os.path.realpath here because we don't want to resolve the symlink
+        spear_data_dir = os.path.join(paks_dir, "SpearData")
+
+        if spear.path_exists(spear_data_dir):
+            print(f"[SPEAR | run_executable.py] File or directory or symlink exists, removing: {spear_data_dir}")
+            spear.remove_path(spear_data_dir)
+
+        print(f"[SPEAR | run_executable.py] Creating symlink: {spear_data_dir} -> {args.data_dir}")
+        os.symlink(args.data_dir, spear_data_dir)
+
+    # provide additional control over which Vulkan devices are recognized by Unreal
+    if len(config.SPEAR.VULKAN_DEVICE_FILES) > 0:
+        print("[SPEAR | run_executable.py] Setting VK_ICD_FILENAMES environment variable: " + config.SPEAR.VULKAN_DEVICE_FILES)
+        os.environ["VK_ICD_FILENAMES"] = config.SPEAR.VULKAN_DEVICE_FILES
+
+    # set up launch executable and command-line arguments
+    launch_args = []
+
+    launch_args.append("-game")
+    launch_args.append("-windowed")
+    launch_args.append("-novsync")
+    launch_args.append("-nosound")
+    launch_args.append("-notexturestreaming")
+    launch_args.append("-resx={}".format(config.SPEAR.WINDOW_RESOLUTION_X))
+    launch_args.append("-resy={}".format(config.SPEAR.WINDOW_RESOLUTION_Y))
+    launch_args.append("-graphicsadapter={}".format(config.SPEAR.GPU_ID))
+
+    if len(config.SPEAR.UNREAL_INTERNAL_LOG_FILE) > 0:
+        launch_args.append("-log={}".format(config.SPEAR.UNREAL_INTERNAL_LOG_FILE))
+   
+    # on Windows, we need to pass in extra command-line parameters to enable DirectX 12
+    # and so that calls to UE_Log and writes to std::cout are visible on the command-line
+    if sys.platform == "win32":
+        launch_args.append("-dx12")            
+        launch_args.append("-stdout")
+        launch_args.append("-fullstdoutlogoutput")
+
+    launch_args.append("-config_file={}".format(temp_config_file))
+
+    for a in config.SPEAR.CUSTOM_COMMAND_LINE_ARGUMENTS:
+        launch_args.append("{}".format(a))
+
+    # launch executable
+    cmd = [executable_internal] + launch_args
+
+    print("[SPEAR | run_executable.py] Launching executable with the following command-line arguments:")
+    print(" ".join(cmd))
+
+    print("[SPEAR | run_executable.py] Launching executable with the following config values:")
+    print(config)
```

Comment:
I think it will be better if we interchange these two sets of print statements. Printing config will be long and will mask the previous print statement, and I think having the launch command as the last print statement will be more helpful.

Reply:
I somewhat disagree, and I'm maintaining consistency with what we already do in env.py